reporter/src/main/kotlin/reporters/fossid/FossIdReporter.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2022 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.reporter.reporters.fossid
+
+import java.io.File
+
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.runBlocking
+
+import org.ossreviewtoolkit.clients.fossid.FossIdRestService
+import org.ossreviewtoolkit.clients.fossid.generateReport
+import org.ossreviewtoolkit.clients.fossid.model.report.ReportType
+import org.ossreviewtoolkit.clients.fossid.model.report.SelectionType
+import org.ossreviewtoolkit.reporter.Reporter
+import org.ossreviewtoolkit.reporter.ReporterInput
+import org.ossreviewtoolkit.utils.core.log
+
+class FossIdReporter : Reporter {
+    companion object {
+        /** Name of the configuration property for the server URL. */
+        const val SERVER_URL_PROPERTY = "serverUrl"
+
+        /** Name of the configuration property for the API key. */
+        const val API_KEY_PROPERTY = "apiKey"
+
+        /** Name of the configuration property for the user name. */
+        const val USER_PROPERTY = "user"
+
+        /** Name of the configuration property for the report type. Default is [ReportType.HTML_DYNAMIC]. */
+        const val REPORT_TYPE_PROPERTY = "reportType"
+
+        /**
+         * Name of the configuration property for the selection type.
+         * Default is [SelectionType.INCLUDE_ALL_LICENSES].
+         */
+        const val SELECTION_TYPE_PROPERTY = "selectionType"
+
+        /**
+         * Name of key in [ScanResult.additionalData] containing the scancode.
+         * TODO: It should be unified with [FossId.SCAN_CODE_KEY], without creating a dependency between scanner and
+         * reporter.
+         */
+        const val SCAN_CODE_KEY = "scancode"
+    }
+
+    override val reporterName = "FossId"
+
+    override fun generateReport(input: ReporterInput, outputDir: File, options: Map<String, String>): List<File> {
+        val serverUrl = options[SERVER_URL_PROPERTY]
+            ?: throw IllegalArgumentException("No FossID server URL configuration found.")
+        val apiKey = options[API_KEY_PROPERTY]
+            ?: throw IllegalArgumentException("No FossID API Key configuration found.")
+        val user = options[USER_PROPERTY]
+            ?: throw IllegalArgumentException("No FossID User configuration found.")
```

Comment:
These should be replaced with

    val serverUrl = requireNotNull(options[SERVER_URL_PROPERTY]) {
        "No FossID server URL configuration found."
    }

etc.